test/buckets/test-create.js

```diff
@@ -28,4 +28,25 @@ describe('buckets/create', () => {
         assert.equal(redirect_url, url_for('buckets.details', { id: created_bucket.id }));
       });
   });
+
+  it('show an error if a bucket already exists with specified name', () => {
+    const form_data = {
+      'new-datasource-name': 'test-bucket',
+    };
+
+    const errors = {
+      name: 's3 bucket with this name already exists.',
+    };
+
+    mock_api()
+      .post('/s3buckets/')
+      .reply(400, errors);
+
+    return dispatch(handlers.create_bucket, { body: form_data })
+      .then(({ redirect_url, template, context }) => {
+        assert.isUndefined(redirect_url);
+        assert.equal(template, 'buckets/new.html');
+        assert.deepEqual(context.error.error, errors);
```

Comment:
May be worth asserting that context passed to template also includes the expected `bucket`/`bucket_prefix` as you mostly have the test already in place.

Reply:
Oops, forgot that